test/smoke/Makefile.rules

```diff
@@ -25,7 +29,11 @@ else ifdef march
 else ifdef make_options
 	$(CC) $(make_options) -o $@
 else
+ifneq (,$(findstring $(INSTALLED_GPU),$(SUPPORTED)))
 	$(SETENV) $(CC) $(CFLAGS) $(EXTRA_CFLAGS) $(OMP_FLAGS) $(wordlist 2,$(words $^),$^) -o $@ $(LINK_FLAGS)
+else
+	@echo "  Skipped compiling $(TESTNAME): supported only on $(SUPPORTED)"
```

Comment:
Do we know (have an informed guess) how well the other machinery, like check-smoke, could `grep` this to do the counting / reporting in the final report it outputs?